PR: https://github.com/odoo/odoo/pull/255566

From: 00e3a35daf35a33a99d04938b66e31a066d3f51e
From: Zelong Lin
Date: 2026-04-01 11:53:35

Breaking data model changes scores: del:3 + add:0, change matches:
-    push_to_talk_key = fields.Char(string="Push-To-Talk shortcut", help="String formatted to represent a key with modifiers following this pattern: shift.ctrl.alt.key, e.g: truthy.1.true.b")
-    use_push_to_talk = fields.Boolean(string="Use the push to talk feature", default=False)
-    voice_active_duration = fields.Integer(string="Duration of voice activity in ms", default=200, help="How long the audio broadcast will remain active after passing the volume threshold")

Total Changes: 4

[IMP] mail: move call settings to local storage

Currently "use_push_to_talk", "push_to_talk_key",
"voice_active_duration" settings are saved on server, which means they
are shared among all devices of the user.

Different devices are likely to use different input methods / keyboards,
therefore these settings should be per device instead.

This commit converts these server settings to localStorage.

task-5992929

closes odoo/odoo#255566

Related: odoo/enterprise#111982
Related: odoo/upgrade#9762
Signed-off-by: Matthieu Stockbauer (tsm) <tsm@odoo.com>

================================= pseudo patch: =================================

--- a/addons/mail/models/res_users_settings.py
+++ b/addons/mail/models/res_users_settings.py
@@ -6,10 +6,6 @@ from odoo import api, fields, models
 class ResUsersSettings(models.Model):
     _inherit = 'res.users.settings'
 
-    # RTC
-    push_to_talk_key = fields.Char(string="Push-To-Talk shortcut", help="String formatted to represent a key with modifiers following this pattern: shift.ctrl.alt.key, e.g: truthy.1.true.b")
-    use_push_to_talk = fields.Boolean(string="Use the push to talk feature", default=False)
-    voice_active_duration = fields.Integer(string="Duration of voice activity in ms", default=200, help="How long the audio broadcast will remain active after passing the volume threshold")
     volume_settings_ids = fields.One2many('res.users.settings.volumes', 'user_setting_id', string="Volumes of other partners")
 
     # Notifications
